ui/src/components/pipelines/pipeline_designer/context_bar/ContextBarSizer.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+
+class ContextBarSizer extends Component {
+  render() {
+    const { changeContextBarSizeFunc, contextBarSize } = this.props;
+
+    const sClassNames =
+      contextBarSize === "S"
+        ? "btn btn-sm btn-dark me-2"
+        : "btn btn-sm btn-outline-dark me-2";
+    const mClassNames =
+      contextBarSize === "M"
+        ? "btn btn-sm btn-dark me-2"
+        : "btn btn-sm btn-outline-dark me-2";
+    const lClassNames =
+      contextBarSize === "L"
+        ? "btn btn-sm btn-dark me-2"
+        : "btn btn-sm btn-outline-dark me-2";
+
+    return (
+      <React.Fragment>
+        <a
+          href="#"
+          className={sClassNames}
+          data-size="S"
```

Comment:
Is there a reason to not use client-side routing for this?

Reply:
From my point of view, it does add any value. Let's replace the hyperlinks with `button`s though